gradle-plugin/src/gradleTest/java/com/squareup/anvil/plugin/AnvilExtensionTest.kt

```diff
@@ -0,0 +1,114 @@
+package com.squareup.anvil.plugin
+
+import com.rickbusarow.kase.kase
+import io.kotest.assertions.asClue
+import io.kotest.matchers.string.shouldContain
+import org.junit.jupiter.api.TestFactory
+
+class AnvilExtensionTest : BaseGradleTest() {
+
+  @TestFactory
+  fun `gradle properties map to extension properties`() = listOf(
+    kase(a1 = "generateDaggerFactories", false),
+    kase(a1 = "generateDaggerFactoriesOnly", false),
+    kase(a1 = "disableComponentMerging", false),
+    kase(a1 = "syncGeneratedSources", false),
+    kase(a1 = "addOptionalAnnotations", false),
+  )
+    .asContainers { (propertyName, default) ->
+      testFactory { versions ->
+
+        rootProject {
+          buildFile(
+            """
+            plugins {
+              kotlin("jvm") version "${versions.kotlinVersion}"
+              id("com.squareup.anvil") version "$anvilVersion"
+            }
+            
+            tasks.register("printProperty") {
+              doLast {
+                println("$propertyName: ${'$'}{anvil.$propertyName.get()}")
+              }
+            }
+            """.trimIndent(),
+          )
+        }
+
+        "the extension property is $default when not set anywhere".asClue {
+
+          shouldSucceed("printProperty") {
+            output shouldContain "$propertyName: $default"
+          }
+        }
```

Comment:
Could you add some helper methods for this test factory? It would make it easier to quickly scan the test for a high-level idea of what is being tested since there's quite a bit going on. E.g. this highlighted chunk could be something like `generateProjectAndVerifyDefaultProperty` or just one for generating the project and leaving the assertion in place.